### DISCUSSION TITLE
PPA GPG Key Expired?

### DISCUSSION THREAD

**Discussion by TaylorAdams** [regolith-linux/regolith-desktop#888] [Q&A] [answered]
Hello, I was just attempting to follow the install instructions in [the docs](https://regolith-desktop.com/) for 20.04, and it looks like there may be an issue with the ppa gpg key?

```W: GPG error: https://regolith-desktop.org/release-ubuntu-focal-amd64 focal InRelease: The following signatures were invalid: EXPKEYSIG BFE717649A5C3D08 Regolith Linux <regolith.linux@gmail.com>
E: The repository 'https://regolith-desktop.org/release-ubuntu-focal-amd64 focal InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
```

Then checking with `apt-key list` it looks like perhaps the key expired today, Aug. 15.  Not sure if that actually, coincidentally is the case or if I just messed up somewhere along the way. Thanks for any tips/assistance.

**Comment by w18g** [comment]
Same on 22.04

```
Err:5 https://regolith-desktop.org/release-ubuntu-jammy-amd64 jammy InRelease
  The following signatures were invalid: EXPKEYSIG BFE717649A5C3D08 Regolith Linux <regolith.linux@gmail.com>

```

**Comment by kgilmer** [comment]
taking a look...

**Reply by kgilmer** [reply]
Seems that I need to sign packages with a new key.  This may take some time.

**Reply by Vosjedev** [reply]
any estimation on how long?

**Comment by ENate** [comment]
Also have same problem. Cannot update with expired key on Ubuntu 22.04

**Comment by anlif** [comment]
a quick hack is to edit /etc/apt/sources.list.d/regolith.list to something like
[arch=amd64 trusted=yes] http:// ...
see https://unix.stackexchange.com/questions/198000/bypass-gpg-signature-checks-only-for-a-single-repository

**Comment by kgilmer** [comment]
There will be no updates to packages in the Regolith repository until the key situation is resolved, so you won't miss anything if you simply disable the repository (by adding a `#` to the front of the line) in the short term, if the errors cause problems:

Example (`/etc/apt/sources.list.d/regolith.list`):
```
# deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] https://regolith-desktop.org/unstable-ubuntu-lunar-amd64 lunar main
```

FYI: I will respond in this thread when a new key has been issued against the repository packages.

**Reply by robgalvan** [reply]
Any suggestions on how to install Regolith on a new device with the expired keys? Tried the `trusted=yes` but that did not work.

**Reply by TaylorAdams** [reply]
For now, adding `allow-insecure=yes` looks like it'll work (credit to the stackexchange link above), at least on 20.04. ie:

```
deb [arch=amd64 allow-insecure=yes] 
```

**Reply by kgilmer** [reply]
If anyone would like to test, the Regolith 3.0 beta has been signed with the new key for Ubuntu Lunar on `amd64`.   The old key name was `regolith.key`.  The new key is called `regolith3.key`, and is (provisionally, until testing is completed) available at https://regolith-desktop.org/regolith3.key.

```
wget -qO - https://regolith-desktop.org/regolith3.key | \
gpg --dearmor | sudo tee /usr/share/keyrings/regolith-archive-keyring.gpg > /dev/null
```

Add unstable Ubuntu Lunar:
```
echo deb "[arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] \
https://regolith-desktop.org/unstable-ubuntu-lunar-amd64 lunar main" | \
sudo tee /etc/apt/sources.list.d/regolith.list
```

All targets in the `unstable` stage are currently being rebuilt with the new key.  Once that's verified, `testing` will be rebuilt next.

NOTE: the info above is intended *only* for testing purposes.  Packages in `unstable` will break from time to time.

**Reply by Vosjedev** [reply]
works for me
```
Hit:1 http://nl.archive.ubuntu.com/ubuntu lunar InRelease
Hit:2 http://dl.google.com/linux/earth/deb stable InRelease                                                                                                            
...  (removed some lines)                                                                                                                                                                                                           
Hit:11 https://dl.winehq.org/wine-builds/ubuntu kinetic InRelease                                                                                                      
Get:12 https://regolith-desktop.org/unstable-ubuntu-lunar-amd64 lunar InRelease [2.379 B]                                                                              
Get:13 https://repo.charm.sh/apt * InRelease                                      
...  (removed some lines) 
Get:31 http://nl.archive.ubuntu.com/ubuntu lunar-backports/universe amd64 DEP-11 Metadata [13,2 kB]
Get:32 https://packages.element.io/debian default/main amd64 Packages [834 B]
Fetched 4.463 kB in 3s (1.522 kB/s)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
7 packages can be upgraded. Run 'apt list --upgradable' to see them.
```

**Reply by kgilmer** [reply]
The key has been updated in-place.  You can reload the key, as directed on the [landing page](https://regolith-desktop.com).  Packages in the `release` stage are building now.  I will update again when the key update work is complete.

**Reply by kgilmer** [reply]
Packages built with the new key completed overnight.  I have a bit of clean up work yet to do but at this point there should be no key issues once the Regolith key has been updated as described on the landing page.  If anyone is still experiencing errors via `apt` please provide details in this thread.

**Reply by lgaboury** [reply]
Thanks Ken for the quick turnaround on this issue.

**Reply by jc00ke** [reply]
Just tested the new key, works great, thanks Ken!!!

**Reply by TaylorAdams** [reply]
Yep, working great, thanks again.

**Reply by ghost** [reply]
Hello, I think a new problem is related with this issue

E: The repository 'https://regolith-desktop.org/release-ubuntu-kinetic-amd64 kinetic Release' no longer has a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.

Can't install any package

**Reply by kgilmer** [reply]
A, yes.  We don't build `kinetic` anymore.  Thanks for reporting @nlopez-backupta.

**Reply by ghost** [reply]
Can you update the doc in https://regolith-desktop.com/ with the new repo for lunar please ? Is it stable now ?

**Reply by kgilmer** [reply]
It hasn't been released yet @nlopez-backupta .  I am working on releasing beta 4 today.

**Reply by kgilmer** [reply]
Installing Regolith 2.2 on Ubuntu Kinetic should be back now.  CC @nlopez-backupta

**Comment by kgilmer** [comment, marked as answer]
Thread summary:

The Regolith key used to sign Debian packages expired.  A new key was generated and all packages were rebuilt with the new key.  The old key was replaced with the new one: `https://regolith-desktop.org/regolith.key`.  

To resolve the "invalid key" error from `apt` when doing an update, install the new key.  This is done by re-running the command listed in the install section of regolith-desktop.com.  For example, on Ubuntu Jammy:

```
wget -qO - https://regolith-desktop.org/regolith.key | \
gpg --dearmor | sudo tee /usr/share/keyrings/regolith-archive-keyring.gpg > /dev/null
```

**Reply by uciharis** [reply]
it works for me ( im using regolith iso kinetic version )

**Reply by harold** [reply]
Also worked great for me - <3 regolith - long may it live. Thank you @kgilmer for everything.

**Reply by snspinn** [reply]
Only getting this error today on Regolith2 (Ubuntu 22.04).

If you're like me and have a habit of using `apt-get update` instead of `apt update`, you'll need to use `apt` to get around the lines:
```
E: Repository 'https://regolith-desktop.org/release-ubuntu-jammy-amd64 jammy InRelease' changed its 'Label' value from 'https://regolith-release-ubuntu-jammy-amd64.s3.us-east-2.amazonaws.com' to 'https://regolith-desktop.org/release-ubuntu-jammy-amd64'
N: This must be accepted explicitly before updates for this repository can be applied. See apt-secure(8) manpage for details.
```

..and then enter "y" to accept this explicitly, when prompted.

**Reply by kgilmer** [reply]
Hi @snspinn , yeah that prompt is due to an earlier change where some decorative metadata was changed in the repo (for awhile S3 was being used to store packages, but we moved away from this to lower costs). You shouldn't have to deal with it again once you've acknowledged the change.
